great_expectations/core/__init__.py

```diff
@@ -900,156 +1635,191 @@ def append_expectation(self, expectation_config):
-            values for discard_result_format_kwargs, discard_include_config_kwargs, and discard_catch_exceptions_kwargs
+        expectation_configuration: ExpectationConfiguration,
+        op: str,
+        path: str,
+        value: Any,
+        match_type: str,
+    ) -> ExpectationConfiguration:
         """
 
-        match_indexes = self.find_expectation_indexes(
-            expectation_type, column, expectation_kwargs,
-        )
+       Args:
+            expectation_configuration: A potentially incomplete (partial) Expectation Configuration to match against to
+                find the expectation to patch.
+            op: A jsonpatch operation (one of 'add','update', or 'remove') (see http://jsonpatch.com/)
+            path: A jsonpatch path for the patch operation (see http://jsonpatch.com/)
+            value: The value to patch (see http://jsonpatch.com/)
+            match_type: The match type to use for find_expectation_index()
 
-        if len(match_indexes) == 0:
-            raise ValueError("No matching expectation found.")
+       Returns: The patched ExpectationConfiguration
 
-        elif len(match_indexes) > 1:
-            if not remove_multiple_matches:
-                raise ValueError(
-                    "Multiple expectations matched arguments. No expectations removed."
-                )
-            else:
+       Raises:
+           No match
+           More than 1 match
 
-                if not dry_run:
-                    self.expectations = [
-                        i
-                        for j, i in enumerate(self.expectations)
-                        if j not in match_indexes
-                    ]
-                else:
-                    return self._copy_and_clean_up_expectations_from_indexes(
-                        match_indexes
-                    )
+               """
+        found_expectation_indexes = self.find_expectation_indexes(
+            expectation_configuration, match_type
+        )
 
-        else:  # Exactly one match
-            expectation = self._copy_and_clean_up_expectation(
-                self.expectations[match_indexes[0]]
+        if len(found_expectation_indexes) < 1:
+            raise ValueError("No matching expectation was found.")
+        elif len(found_expectation_indexes) > 1:
+            raise ValueError(
+                "More than one matching expectation was found. Please be more specific with your search "
+                "criteria"
             )
 
-            if not dry_run:
-                del self.expectations[match_indexes[0]]
+        self.expectations[found_expectation_indexes[0]].patch(op, path, value)
+        return self.expectations[found_expectation_indexes[0]]
+
+    def add_expectation(
+        self,
+        expectation_configuration: ExpectationConfiguration,
+        match_type: str = "domain",
+        overwrite_existing: bool = False,
```

Comment:
Shouldn't the default for `overwrite_existing` be `True`, to be in line with current behavior?

Reply:
Probably! But I wasn't 100% sure and wanted to be safe. I'll change it.